dv_utils/client.py

```diff
@@ -16,18 +17,71 @@
 
 class Client:
     """
-    Http client to interact with the Data-village API.
+    Http client to interact with the Datavillage API.
     """
+    DATA_PROVIDER_COLLABORATOR_ROLE_VALUE="DataProvider"
+
+    DATA_PROVIDER_COLLABORATOR_ROLE_VALUE="DataProvider"
+    DATA_CONSUMER_COLLABORATOR_ROLE_VALUE="DataConsumer"
 
     def __init__(self, settings: Settings = None):
         self.settings: Settings = settings or default_settings
+    
+    def get_data_descriptor(self, collaboration_space_id: str, data_descriptor_id: str, role: str):
+        """
+        Returns the data descriptor as json 
+        """
+        try:
+            list_participants = self.request(
+                f"/collaborationSpaces/{collaboration_space_id}/collaborators"
+            )
+            if role!=None:
+                filtered_participants = [x for x in list_participants if x["role"]==role]
+            else:
+                filtered_participants = [x for x in list_participants]
+            data_descriptors = [ 
+                descriptor for participant in filtered_participants    
+                for descriptor in participant.get("dataDescriptors", [])
+                if descriptor["id"] == data_descriptor_id
+                ]
+            return data_descriptors[0]
+        except Exception as e:
+            logger.error(e)
+            return None
+    
+    def get_data_descriptors_for_collaboration_space(self, collaboration_space_id: str, role: str):
+        """
+        Returns the data descriptors as array of json 
+        """
+        list_participants=self.get_list_of_participants(collaboration_space_id,role)
+        if list_participants != None and len(list_participants)>0:
+            for participant in list_participants:
+                client_id=participant["clientId"]
+                data_descriptors=participant.get("dataDescriptors")
+
+    def get_list_of_participants(self, collaboration_space_id: str, role: str):
+        """
+        Returns the participants as json 
+        """
+        try:
+            list_participants = self.request(
+                f"/collaborationSpaces/{collaboration_space_id}/collaborators"
+            )
+            if role!=None:
+                filtered_participants = [x for x in list_participants if x["role"]==role]
+            else:
+                filtered_participants=list_participants
+            return filtered_participants
+        except Exception as e:
+            logger.error(e)
+            return None
```

Comment:
not sure which exceptions you try to handle here, but if it's network-related, maybe we should handle those in the `self.request()` method?